src/main/common/filter.c

```diff
@@ -117,7 +77,67 @@ float rateLimitFilterApply4(rateLimitFilter_t *filter, float input, float rate_l
     return filter->state;
 }
 
-// FIR filter
+float filterGetNotchQ(uint16_t centerFreq, uint16_t cutoff)
```

Comment:
This PR just adds the notch filter code from betaflight. It doesn't actually use it anywhere. We need some more work to set up some CLI and MSP variables to set the values and also we need to change the PID and gyro code to actually use it. But this is a start.